server/image_pull.go

```diff
@@ -228,6 +239,37 @@ func (s *Server) pullImage(ctx context.Context, pullArgs *pullArguments) (string
 	return imageRef, nil
 }
 
+func tryIncrementImagePullFailureMetric(ctx context.Context, img string, err error) {
+	// We try to cover some basic use-cases
+	const labelUnknown = "UNKNOWN"
+	label := labelUnknown
+
+	// Docker registry errors
+	for _, desc := range errcode.GetErrorAllDescriptors() {
+		if strings.Contains(err.Error(), desc.Message) {
+			label = desc.Value
+			break
+		}
+	}
+	if label == labelUnknown {
+		if strings.Contains(err.Error(), "connection refused") { // nolint: gocritic
+			label = "CONNECTION_REFUSED"
+		} else if strings.Contains(err.Error(), "connection timed out") {
+			label = "CONNECTION_TIMEOUT"
+		} else if strings.Contains(err.Error(), "404 (Not Found)") {
+			label = "NOT_FOUND"
+		}
+	}
+
+	// Update metric for successful image pulls
```

Comment:
comment should be for failed image pulls.

Reply:
Makes sense, fixed that comment :+1: